File: lecture_12/p6.java
public class Solution {
    /**
     * @param A: lists A
     * @param B: lists B
     * @return: the index mapping
     */
    public int[] anagramMappings(int[] a, int[] b) {
        int[] ans=new int[a.length];
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<b.length;i++)
            map.put(b[i],i);
        for(int i=0;i<a.length;i++)
        {
            ans[i]=map.get(a[i]);
        }
        return ans;
    }
}

File: lecture_39/p6.cpp
class Solution {
public:
    /**
     * @param grid: a list of lists of integers
     * @return: return an integer, denote the number of distinct islands
     */
    void dfs(vector<vector<int>> &grid,int r,int c,string& s)
    {
        grid[r][c]=0;
        if(c+1<grid[0].size() && grid[r][c+1]==1)
        {
            s+="r";
            dfs(grid,r,c+1,s);
        }
        if(r+1<grid.size() && grid[r+1][c]==1)
        {
            s+="d";
            dfs(grid,r+1,c,s);
        }
        if(c-1>=0 && grid[r][c-1]==1)
        {
            s+="l";
            dfs(grid,r,c-1,s);
        }
        if(r-1>=0 && grid[r-1][c]==1)
        {
            s+="u";
            dfs(grid,r-1,c,s);
        }
        s+="x";
    }
    int numberofDistinctIslands(vector<vector<int>> &grid) {
        unordered_set<string> isTaken;
        for(int i=0;i<grid.size();i++)
        {
            for(int j=0;j<grid[i].size();j++)
            {
                if(grid[i][j]==1)
                {
                    string s="";
                    dfs(grid,i,j,s);
                    isTaken.insert(s);
                }
            }
        }
        return isTaken.size();
    }
};